obs/project.go

```diff
@@ -40,13 +40,98 @@ type Project struct {
 	UseForBuild  *UseForBuild `json:"useForBuild,omitempty" xml:"useforbuild,omitempty"`
 }
 
-type Client struct {
-	Client   http.Client
+// OBS is the wrapper struct
+type OBS struct {
+	client  Client
+	options *Options
+}
+
+type obsClient struct {
+	*http.Client
+	*Options
```

Comment:
Options are not needed here

Reply:
InvokeOBSEndpoint has a receiver of type `obsClient` and we need to access the `username` and `password` to set the authentication. Hence this is the reason why Options have been included here.

Reply:
yeah, i think that should be a parameter as well

also, did you create a small testing cli code to use this and test it manually to check if you were able to do things?